### PR TITLE
harmonize color input

### PR DESCRIPTION
**Pull request by CorentinSardo**
### Step 1 :
Every input should be with text
### TODO :
Bubble & Network

### CODE REVIEW

**Review by touv — CHANGES_REQUESTED** [review]
I think you should read this blog :
 https://kentcdodds.com/blog/please-dont-commit-commented-out-code

**Comment by CorentinSardo** [comment]
Not really sure it's fine testing (it doesn't use mapStateToProps), but tests are now valid and the same checks are done.
In fact, colors in text mode aren't tested yet in the app, so I had to try to do it by myself.

**Comment by CorentinSardo** [comment]
Shall we factorize the code color input code in this PR or in the next one ?

File: src/app/js/formats/title/TitleAdmin.js [hunk 1]
```diff
@@ -21,7 +21,7 @@ const styles = {
 
 export const defaultArgs = {
     level: 1,
-    textColor: '#000',
+    textColor: '#818181',
```
Comment (parmentf) [inline src/app/js/formats/title/TitleAdmin.js]: The idea is neat, but I would have used a constant (defined in an external file, perhaps within src/app/js/formats).

File: src/app/js/formats/hierarchy/Hierarchy.js [hunk 2]
```diff
@@ -300,7 +284,12 @@ class Hierarchy extends PureComponent {
                         d.parent.y
                     );
                 })
-                .attr('stroke', color);
+                .attr(
+                    'stroke',
+                    /(^#[0-9A-F]{6}$)|(^#[0-9A-F]{3}$)/i.test(color)
```
Comment (parmentf) [inline src/app/js/formats/hierarchy/Hierarchy.js]: This limits the color to hexadecimal codes.
Maybe we should create a function `isValidColor` and use it as is now.
That way, we could improve the function later.
And the function could be common to all color parameters.

File: src/app/js/formats/bubbleChart/BubbleView.js [hunk 3]
```diff
@@ -48,7 +48,13 @@ export const BubbleView = ({ data, diameter, colorSet }) => (
                     y={y}
                     name={key}
                     value={value}
-                    color={colorSet[index % colorSet.length]}
+                    color={
+                        /(^#[0-9A-F]{6}$)|(^#[0-9A-F]{3}$)/i.test(
```
Comment (parmentf) [inline src/app/js/formats/bubbleChart/BubbleView.js]: Same code here.
Definitely, time for a function (see previous comment).

**Review by parmentf — CHANGES_REQUESTED** [review]

**Comment by parmentf** [comment]
Oh. It seems it was already present in original code, but `eslint(react/prop-types)` rule is triggered within `Hierarchy.js`, and likely in other formats too.

One day, we'll have to improve that.

File: src/app/js/formats/bubbleChart/BubbleView.js (hunk 3 above)
Comment (CorentinSardo) [reply src/app/js/formats/bubbleChart/BubbleView.js]: Definitely better that way, external file [src/app/js/formats/colorUtils.js](https://github.com/Inist-CNRS/lodex/blob/293798c30299e4ce54b8d594d9c4c31034cc2977/src/app/js/formats/colorUtils.js) added

**Pull request merged**